[Colin] iP

cowlinn opened the pull request:
### **Have no fear, DukeBot is here** 

> _"Many users say that DukeBot just might be the greatest invention in the 21st century"_ - Source: `https://www.youtube.com/watch?v=dQw4w9WgXcQ

This bot is a simple task management bot, it has the following plus points

- Easy to use 
- Cool Colored text boxes 🟦 
- Free! 🆓 

### Additionally, you can use the bot in a few easy steps 

1. Download it from [here ](https://github.com/cowlinn/ip/releases/tag/v0.2)
2. Let it aid in your procrastination!

Features:

- [x] Manages various types of tasks 
- [x] Finds tasks by keywords 
- [x] Add/Delete/Mark/Ummark
- [x] Maintain an ordered list of tasks by date(WIP! :<)

AND MANY MORE!

## Catch some cool exceptions! 
If you wanna practice some exception handling, here's a sample `exception`
```

    public EmptyTaskException(String message) {
        super(message);
    }
```

Commit:
Add Gradle support

Commit:
Level - 1

Added Greet, Echo, Exit

Commit:
Level-2

Added Add,List functionalities

Commit:
Level 3

Added Mark As Donne and Tasks

Commit:
Level - 4

Added: ToDo, Event (as subclasses of Tasks), Deadline

Commit:
A-TextUI Testing & Level-5

I/O text files changed + error handling

Commit:
Level 6

Added delete + tidied up error handling

Commit:
Level-7

Added save

Commit:
Level 8

Changed individual task classes to support datetime

Commit:
Updated save logic

Commit:
More changes to save

cowlinn committed:
Merge pull request #1 from cowlinn/branch-Level-7

Branch level 7

Commit:
Merge branch 'master' into branch-Level-8

cowlinn committed:
Merge pull request #2 from cowlinn/branch-Level-8

Branch level 8

Commit:
A-MoreOOP

Added more OOP classes:
UI, Storage, Parser, TaskList

Commit:
Changed Save Data location

Saved binary SaveData in a new folder

Commit:
A-Packages

Added packages
Added new main class to execute program

cowlinn committed:
Merge pull request #3 from nus-cs2103-AY2223S1/add-gradle-support

Test Add Gradle Support from upstream brunch

Commit:
Added JUnit Tests

Tested 2 classes and their methods
Added Main method in duke.Duke

Commit:
Built JAR file

Updated mainClassName Class path
Still unable to save

Commit:
Update Storage.java

Updated save to create a directory + file + other error handling

Commit:
Level-9

Added feature to find

Commit:
Updated coding standard

Take these as the correct ones upon merge

cowlinn committed:
Merge pull request #4 from cowlinn/branch-Level-9

Level-9

Commit:
Merge branch 'master' into brach-A-CodingStandard

cowlinn committed:
Merge pull request #5 from cowlinn/brach-A-CodingStandard

Branch A coding standard

Commit:
Added Find functionality

Fixed bug where find functionality wasn't working properly, probably overriden

Commit:
Fix Checkstyle Issues

Fix Checkstyle-related issues

Commit:
Re-Added javadoc comments

Added Javadoc comments to the classes Duke, Event (and sub-classes of event) UI and Parser, as they contain the main functionalities of the class

Commit:
Cleaned up Javadocs

cowlinn committed:
Merge pull request #6 from cowlinn/branch-A-JavaDoc

Branch A - java doc

Commit:
Updated Build.gradle

Included javafx dependencies

Commit:
Added sample GUI

TODO: Consolidate message into string

Commit:
Added the main logic for the GUI

TODO: Exit properly / flip portraits

Commit:
Converted GUI Scene and logic using FXML

Added FXML files MainWindow and DialogBox. Removed UI visual tweaks and Utility method from the main Duke file.

Commit:
Changed formatting methods access modifiers

Trying to adhere to Abstraction by removing some publicly exposed behavior.

Commit:
Removed unnecessary code from Duke.main

Removed the need to initialize anything from main function in duke

Commit:
Styling for Dialogue Box and Icon

Changed the styles for the dialogue Box and Icon to be more aesthetically pleasing.

cowlinn committed:
Merge pull request #7 from cowlinn/branch-Level-10

Branch level 10

Commit:
Added SaveData Binary file to GitIgnore

Commit:
Updated the main class to Launcher instead of Main in build.gradle

Small change to mainClassName to launch from duke.Launcher instead of the main class, which doesn't exist anymore.

Commit:
Tidying up Storage.java

- Tidied up Storage.java to not use wildcard imports

Commit:
Changed SaveData to human readable format

General bug-fixing and style changes as well

Commit:
General Bug fixes

Removed some deprecated methods

Commit:
Added Assertions to Commands

Input validation for arrays passed into the 3 commands

Commit:
Refactored Code

Added more distinct packages to show the difference in names

Commit:
Refactoring into separate packages

Removed some print debugging statements
Added SaveData folder to git-ignore
Further segment code into additional packages: main/parser/ui

Commit:
Merge branch 'master' into branch-A-Assertions

Commit:
Added functionality to view events that occurred at a given date

Parser class is now able to parse a given input string and match it with valid events that occurred on that day.

Also did general improvements for javadocs

cowlinn committed:
Merge pull request #8 from cowlinn/branch-A-Assertions

Creating PR for A-Assertions

Commit:
Merge branch 'master' into branch-B-ViewSchedules

cowlinn committed:
Merge pull request #9 from cowlinn/branch-B-ViewSchedules

Merge Feature: viewSchedules

Commit:
Added a help-page

When the user types in the word "help" into the chatbot, a google document containing all the appropriate commands and syntax will appear

phyliciachristel commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,88 @@
+     * Used when index is out of bounds for add/delete
+     * @return formatted Out of bounds message string
+     */
+    public static String outOfBoundsMessage(int upperLimit, int index) {
```
Maybe this method name should be a verb instead?

cowlinn replied:
Hey good catch there thank you! Will change the Javadocs and method spacing issues as well

phyliciachristel commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,88 @@
+
+    /**
+     * Returns the number of tasks left
+     * @param tasksLeft
```
Perhaps you should add an empty line between the description and the parameter sections. I noticed this in other parts of your code as well.

phyliciachristel commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,88 @@
+     * Prints it out nicely with the appropriate numbering & indentation
+     * @param taskList
+     */
+
```
Would it be better if you didn't leave a line between the documentation block and the start of the method? I noticed this in other parts of your code as well.

phyliciachristel commented on src/main/java/duke/events/Event.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+
+    public static Event readTask(String[] values) {
```
Would it be better if you added some javadocs here?

Commit:
Trivial update

Small bugfixing before merging with main
Changed the main background color of JavaFX

cowlinn committed:
Merge pull request #10 from cowlinn/branch-C-Help

Merge Branch C into main

Commit:
Included switch statement in parser

Improved parser logic to parse the first input command

DarrenCsAcc reviewed (commented):
Good job with the naming

DarrenCsAcc reviewed (commented):
.

Commit:
Update JavaDocs and general cleanup

JavaDoc methods are too short and undescriptive

The parameter tags are often missing or one-liners at best.

Other users reading the code might not have sufficient idea as to what is going on with the code.

Let's update the JavaDocs in line with recommendations from the textbook.

As well as checkstyle to provide more description, even in helper methods.

Doing so allows readers to understand the codebase better at a glance

Commit:
Minor code refactor

Update some method names to verb form

Commit:
Minor updates improvements for Parser

Reduced redundant code for index related tasks

More informative error calls for "help" function

Commit:
Update Parser's JavaDocs

Commit:
Change naming conventions in Parser.java and Storage.java

Parser and Storage classes contain a lot of low-level functionalities

Intermediary variables are currently named inappropriately

Providing better naming conventions improves the overall readability

And removing magic numbers gives meaning to constants throughout the codebase

Let's refactor the naming scheme for the low-level classes

Into names that are better appreciated by developers
And provide some meaning to the variable's purpose and type

Commit:
Add Checkstyle and update build

Added checkstyle and fixed some spacing/misc. errors

cowlinn committed:
Added Test.YML in CI

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Added more unit testing

Updated to test TaskGeneration and viewSchedule methods after major refactoring

Commit:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Checkstyle fixes

More checkstyle and formatting issues
Changed parser to abstract out methods to adhere to SLAP

Commit:
Revert "Checkstyle fixes"

This reverts commit 141bbe5e8129ea059e831190a448485c76a57243.

Commit:
Revert "Merge branch 'master' into branch-A-CodeQuality"

This reverts commit adadb596bdd1950a0711e62256c7dd1de5de8de4, reversing
changes made to 5bb51ace1d90ff289decaf170c3c55e88a34224e.

Commit:
Revert "Merge branch 'master' into branch-A-CodeQuality"

This reverts commit 5bb51ace1d90ff289decaf170c3c55e88a34224e, reversing
changes made to b31f17dc8492c699689d0f151eb17ffd7f64425d.

Commit:
Revert "Add Checkstyle and update build"

This reverts commit b31f17dc8492c699689d0f151eb17ffd7f64425d.

Commit:
Updated parser

Now all types of words should be allowed to be parsed in

Commit:
Squashed bugs in parser and dialogueBox

Fixed some minor bugs where inputs were not allowed to be passed in / images were not properly displayed.

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Revert Checkstyle original changes

Commit:
Test ReadME.md

Just a test if it updates

Commit:
Updated ReadME

Commit:
Updated Readme

Includes basic functionality of website

Commit:
Updated ReadME and cleaned up headers

Commit:
Updated ReadMe

Commit:
Changed formatting of ReadMe

Minor bug fixes in display orientation

Commit:
Merge branch 'master' of https://github.com/cowlinn/ip

Commit:
Tidy up Product Website

Updated the correct ReadME in docs, added screenshot

Commit:
Update App Icon and Name

Updated Ui page as well, will need to add credits soon

Commit:
Update Acknowledgements in ReadMe

Commit:
Changed DukeIcon

Added royalty free version of new icon for duke from OGA

Commit:
Update ReadME

Updated resources to include free images, added credits and updated Ui

Commit:
Fix Save Bug

Fix bug where file would not be created properly

General changes formatting and style changes

Commit:
Minor bugfixes

Fixed ReadMe bugs and added a standard font to view user input (bugfix for MacOS)

Commit:
Updated formatting for ReadME

Updated tagging

Commit:
Added Spacing/formatting for ReadMe

Commit:
General Bug Fixes

Fixed window size to prevent resizing

Allowed entire Task view to be seen where previously, only a small segment was visible.

Updated build JavaFX Version to allow app to run in newer Ubuntu

Handled exceptions for incorrect number of arguments passed in for event/deadline

Commit:
Removed  old checkstyle files no longer in use
